edstem-api/edstem_api/macros.py

```diff
@@ -0,0 +1,233 @@
+""""""
```

Comment:
Again, we will want to eventually document all of these functions. Did you find most were self explanatory or that you tended to struggle with these having to do lots of tests?

Reply:
To be completely honest, I forgot that your changes didn't have comments lol. I can start adding some if I understand the function, and we can rebase + recommit to the PR

Reply:
It was more a question so that when I edit docstrings later, I know how in depth I should actually go into the process of explanation. I'm thinking most of these are decently obvious but the expected types are ambiguous right now.

Reply:
Yeah, maybe a one liner for the description